src/krux/pages/qr_view.py

```diff
@@ -382,9 +382,56 @@ def save_bmp_image(self, file_name, resolution):
         self.ctx.display.draw_centered_text(t("Processing.."))
 
         bmp_img.save("/sd/" + file_name)
-        self.flash_text(
-            t("Saved to SD card:") + "\n%s" % file_name, highlight_prefix=":"
+        self.flash_text(t("Saved to SD card:") + "\n%s" % file_name)
+
+    def save_svg_image(self, file_name):
+        """Save QR code image as .svg file"""
+        from ..sd_card import SVG_IMAGE_EXTENSION
+        from .file_operations import SaveFile
+
+        scale = 10
+
+        code, size = self.add_frame(self.code, self.qr_size)
+
+        save_page = SaveFile(self.ctx)
+        file_name = save_page.set_filename(
+            file_name, file_extension=SVG_IMAGE_EXTENSION
         )
+        if file_name == ESC_KEY:
+            return
+
+        self.ctx.display.clear()
+        self.ctx.display.draw_centered_text(t("Processing.."))
+
+        with open("/sd/" + file_name, "w") as f:
+            width = size * scale
+            height = width
+            f.write(
+                '<svg xmlns="http://www.w3.org/2000/svg" width="{}" height="{}">\n'.format(
+                    width, height
+                )
+            )
+            for y_index in range(0, size):
+                for x_index in range(0, size):
+                    index = y_index * size + x_index
+                    bit_value = (code[index >> 3] >> (index % 8)) & 1
+                    if bit_value:
+                        x = x_index * scale
+                        y = y_index * scale
+                        scale_x = scale
+                        scale_y = scale
+                        square = 'x="{}" y="{}" width="{}" height="{}"'.format(
+                            x, y, scale_x, scale_y
+                        )
+                        f.write(
+                            '<rect stroke="black" stroke-width="0" {} fill="black"/>\n'.format(
+                                square
+                            )
+                        )
+            f.write("</svg>")
+            f.close()
```

Comment:
Here it doesn't need to call f.close() because it is inside a `with` statement, so it will call `flush` and `close` automatically